pkg/apiserver/handlers.go

```diff
@@ -534,3 +534,28 @@ func (r *APIRequestInfoResolver) GetAPIRequestInfo(req *http.Request) (APIReques
 
 	return requestInfo, nil
 }
+
+// IsAPIRequest looks for patterns like /prefix/version/resource and /prefix/group/version/resource. It if matches, returns true.
```

Comment:
Why not also do the extraction and return these things? I don't like having extraction and predicate code possibly evolving separately.